sql/api/src/main/antlr4/org/apache/spark/sql/catalyst/parser/SqlBaseParser.g4

```diff
@@ -1096,7 +1096,7 @@ colPosition
     ;
 
 collateClause
-    : COLLATE collationName=stringLit
+    : COLLATE collationName=identifier
```

Comment:
BTW, not `errorCapturingIdentifier`? Do you have a test for `... COLLATE BLA_BLA-1`

Reply:
why would we want to capture `BLA_BLA-1`? afaik immediate follow symbol can be an expression and thus `errorCapturingIdentifier` should not be used right?

Reply:
Let's look at the examples:
```
spark-sql (default)> select 'aaa' collate test_collation;
[COLLATION_INVALID_NAME] The value test_collation does not represent a correct collation name. Suggested valid collation name: [UCS_BASIC]. SQLSTATE: 42704
```
The error message says about `test_collation`, that's correct but:
```
spark-sql (default)> select 'aaa' collate test-collation;
[COLLATION_INVALID_NAME] The value test does not represent a correct collation name. Suggested valid collation name: [UNICODE]. SQLSTATE: 42704
```
it points out `test`.

Reply:
BTW, I would improve the error message to:
```
[COLLATION_INVALID_NAME] `test_collation` does not represent a correct collation name. Suggested valid collation name: UCS_BASIC. SQLSTATE: 42704

```

Reply:
what about this case:
```code
select 'a' collate UNICODE-columnB
```

then the error would be:
```code
[INVALID_IDENTIFIER] The identifier UNICODE-columnB is invalid. Please, consider quoting it with back-quotes as `UNICODE-columnB`. SQLSTATE: 42602
```

instead of giving the error that types are incompatible for minus operation

Reply:
I agree, it would be better to report about incompatible types rather invalid identifier.